### PR TITLE
use web audio buffers for sound instead of capacitor plugin

### PR DESCRIPTION
**Pull request by schlawg**
See #10809

The sound for lichess app has not been working for wired (lightning) headphones since at least iOS 15 and maybe longer.  Those headphones came with every iPhone sold up until 2 years ago, so there's a lot of them out there.  Strangely, bluetooth audio was unaffected.  I removed the capacitor audio plugin stuff which was garbage time, and replaced it with buffered web audio so effects now always play and they play instantly.  Also got rid of the aifc files.  Tested on an iPhone SE with iOS 15.5 and android 12/S via Pixel XL emulator.

### CODE REVIEW

**Comment by schlawg** [comment]
to clarify, that's #10809 in the lila base.  He logged it as a lila issue but it's a lichobile bug.

https://github.com/lichess-org/lila/issues/10809

**Comment by veloce** [comment]
I can reproduce the issue. I'll merge the fix asap but I need to test it on a real android device first, and I can't do that before 1 week.

**Comment by schlawg** [comment]
Cool.  Let me know if want me to package those buffers into a tidy object or make the sound.ts file more "lichobley" in some way.  Ordinarily I'd take the time to do that kind of stuff up front but (hopefully) this code will have a pretty short lifetime.  I just wanted to fix it and be done.

File: src/sound.ts [hunk 1]
```diff
@@ -1,59 +1,92 @@
...
-      SoundEffect.loadSound({ id: 'berserk', path: `sounds/berserk${ext}` }),
-      SoundEffect.loadSound({ id: 'clock', path: `sounds/clock${ext}` }),
-      SoundEffect.loadSound({ id: 'confirmation', path: `sounds/confirmation${ext}` }),
-    ]).then(() => { /* noop */ })
-  },
-  move() {
-    if (shouldPlay()) SoundEffect.play({ id: 'move' })
+  load(): Promise<void> {
+    if ( ctx == null ) {
+      return new Promise<void>(() => {}) // hoping this fuckery makes jest happy
```
[the remark this answers is not in the document]
Comment (schlawg) [reply src/sound.ts:10]: Promise.resolve() works fine.  The ctx==null and empty promises are just me trying to satiate ForecastCtrl unit test (which loads the sound object without access to window object).  I didn't want to modify anything about the test setup.  It was late and I just wanted the lint to pass.  :)

I figured this code will have a short lifetime so I didn't exactly go out of my way to look up the best way to do things or fit it in with the rest of lichobile.  Please feel free to make any edits you want!  Or just request them and I will enact.

**Comment by veloce** [comment]
Sorry I can't merge it as is. This PR introduces a regression as the silent switch on iOS devices has no effect anymore.

**Comment by schlawg** [comment]
The silent switch works fine on my iPhone SE.  jas14 how about you?

**Comment by jas14** [comment]
I only have Android 😬 I'm happy to test on it, though!

**Comment by schlawg** [comment]
Veloce, the silent switch turns off all audio at the OS level.  There is absolutely no way that application code could tunnel through it.

**Comment by veloce** [comment]
@jas14 you can test on android if you want, basically it should be non-regression tests while checking: silent mode, volume, how the sound mix with a background music, etc.

As for the silent switch on iOS it's not working on my simulator but unfortunately I was unable to test on a real device so far, I'll try again as soon as I can.

File: src/sound.ts [hunk 2]
```diff
@@ -1,59 +1,87 @@
...
+        loadSound('lowtime').then(buf => (lowtime = buf)),
+        loadSound('dong').then(buf => (dong = buf)),
+        loadSound('berserk').then(buf => (berserk = buf)),
+        loadSound('clock').then(buf => (clock = buf)),
+        loadSound('confirmation').then(buf => (confirmation = buf)),
+      ]).then(() => console.log('all sounds loaded.'))
+  },
+  
+  resume(): void {
+    if (ctx != null ) {
```
Comment (veloce) [inline src/sound.ts:24]: don't understand this. why not use `AudioContext.suspend()` and `AudioContext.resume()` instead?

**Comment by veloce** [comment]
Update:
- I changed the code to use the plugin on android and older iOS versions (AudioContext API doesn't work on iOS <14.5)
- I checked on an iPhone using wired headphones and it's good
- quickly tested on android to check there is no regression
- I haven't tested older iOS versions, hope it's fine

**Comment by schlawg** [comment]
Good catch on iOS < 14.5!  I think capacitor worked for those versions anyway, so hopefully this is a total fix.

**Pull request merged**